tests/integration/security/sds_tls_origination/util/util.go

```diff
@@ -0,0 +1,410 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sdstlsutil
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"html/template"
+	"io/ioutil"
+	"path"
+	"testing"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"istio.io/istio/pkg/test"
+	"istio.io/istio/pkg/test/env"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/environment/kube"
+	"istio.io/istio/pkg/test/framework/components/istio"
+
+	"istio.io/istio/pkg/test/echo/common"
+	"istio.io/istio/pkg/test/framework/resource"
+
+	envoyAdmin "github.com/envoyproxy/go-control-plane/envoy/admin/v3"
+
+	"istio.io/istio/pkg/config/protocol"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/echoboot"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/util/retry"
+	"istio.io/istio/pkg/test/util/structpath"
+)
+
+func MustReadCert(t *testing.T, f string) string {
+	b, err := ioutil.ReadFile(path.Join(env.IstioSrc, "tests/testdata/certs/dns", f))
+	if err != nil {
+		t.Fatalf("failed to read %v: %v", f, err)
+	}
+	return string(b)
+}
+
+type TLSCredential struct {
+	PrivateKey string
+	ClientCert string
+	CaCert     string
+}
+
+const (
+	// The ID/name for the certificate chain in kubernetes tls secret.
+	tlsScrtCert = "tls.crt"
+	// The ID/name for the k8sKey in kubernetes tls secret.
+	tlsScrtKey = "tls.key"
+	// The ID/name for the CA certificate in kubernetes tls secret
+	tlsScrtCaCert = "ca.crt"
+	// The ID/name for the certificate chain in kubernetes generic secret.
+	genericScrtCert = "cert"
+	// The ID/name for the private key in kubernetes generic secret.
+	genericScrtKey = "key"
+	// The ID/name for the CA certificate in kubernetes generic secret.
+	genericScrtCaCert = "cacert"
+)
+
+// CreateKubeSecret reads credential names from credNames and key/cert from TLSCredential,
+// and creates K8s secrets for gateway.
+// nolint: interfacer
+func CreateKubeSecret(t test.Failer, ctx framework.TestContext, credNames []string,
```

Comment:
There is a CreateIngressKubeSecret() in tests/integration/security/sds_ingress/util/util.go, could you refactor that one and use it here, instead of adding a similar util function?

Reply:
see comment above